stix2viz/stix2viz/stix2viz.js

```diff
@@ -779,6 +800,64 @@ function makeEdgeObject(sourceRef, targetRef, label, stixId=null)
     return edge;
 }
 
+const timelineTimestamps = {                                   
+    "attack-pattern": ["modified", "created"],
+    "campaign": ["last_seen", "first_seen", "modified", "created"],
+    "course-of-action": ["modified", "created"],
+    "event": ["start_time", "modified", "created"],
+    "identity": ["modified", "created"],
+    "task": ["start_time", "modified", "created"],
+    "incident": ["modified", "created"],
+    "indicator": ["modified", "created"],
+    "infrastructure": ["modified", "created"],
+    "intrusion-set": ["last_seen", "first_seen", "modified", "created"],
+    "location": ["modified", "created"],
+    "malware": ["last_seen", "first_seen", "modified", "created"],
+    "malware-analysis": ["modified", "created"],
+    "note": ["modified", "created"],
+    "observed-data": ["last_observed", "first_observed", "modified", "created"],
+    "opinion": ["modified", "created"],
+    "report": ["modified", "created"],
+    "task": ["start_time", "modified", "created"],
+    "threat-actor": ["last_seen", "first_seen","modified", "created"],
+    "tool": ["modified", "created"],
+    "vulnerability": ["modified", "created"],
+
+    "relationship": ["start_time", "modified", "created"],
+    "sighting": ["last_seen", "first_seen", "modified", "created"]
+};
+
+function determineTimestampFromConfig(stixObject, stixType, config)
+{
+    if (config.has(stixType)) {
+        let typeConfig = config.get(stixType);
+        if (typeConfig.has("timestampList"))
+            return determineTimestamp(stixObject, typeConfig.get("timestampList"), null);
```

Comment:
If the config structure is changed, the config docs should be updated:

https://github.com/oasis-open/cti-stix-visualization/blob/master/README.Javascript.rst#configuration